maro/simulator/scenarios/data_center/business_engine.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from copy import deepcopy
+from typing import List
+
+from yaml import safe_load
+
+from maro.event_buffer import AtomEvent, CascadeEvent, EventBuffer, MaroEvents
+from maro.simulator.scenarios.abs_business_engine import AbsBusinessEngine
+from maro.simulator.scenarios.helpers import DocableDict
+
+from .common import Action, Latency, Payload
+from .events import Events
+from .physical_machine import PhysicalMachine
+from .virtual_machine import VirtualMachine
+
+metrics_desc = """
+
+energy_consumption (int): Current total energy consumption.
+success_requirements (int): Accumulative successful VM requirements until now.
+total_latency (int): Accumulative buffer time until now.
+"""
+
+
+class DataCenterBusinessEngine(AbsBusinessEngine):
+    def __init__(
+        self, event_buffer: EventBuffer, topology: str, start_tick: int,
+        max_tick: int, snapshot_resolution: int, max_snapshots: int, additional_options: dict = {}
+    ):
+        super().__init__(
+            "data_center", event_buffer, topology, start_tick, max_tick,
+            snapshot_resolution, max_snapshots, additional_options
+        )
+
+        self._energy_consumption: int = 0
+        self._success_requirements: int = 0
+        self._total_latency: Latency = Latency()
+
+        self._init()
+
+    def _load_configs(self):
+        """Load configurations"""
+        with open(os.path.join(self._config_path, "config.yml")) as fp:
+            self._conf = safe_load(fp)
+
+    def _init(self):
+        # load config
+        self._load_configs()
+        self._register_events()
+        # PM init
+        self.machines: List[PhysicalMachine] = []
+        pm_amount = self._conf["pm_amount"]
+        cap_cpu = self._conf["pm_cap_cpu"]
+        cap_mem = self._conf["pm_cap_mem"]
+        self.machines = [PhysicalMachine(id=i, cap_cpu=cap_cpu, cap_mem=cap_mem) for i in range(pm_amount)]
```

Comment:
if line length exceeds, you can wrap lines according to ``Hanging Grid Grouped`` like:
![image](https://user-images.githubusercontent.com/53509467/98495738-d3d15d00-227a-11eb-907a-70a80a41feb0.png)